### ISSUE TITLE
Update ECMPS to Import daily backstop [GNP] (emissions) data

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#5395]
## Context 

As apart of the scope of this ticket ensure users can import the new schema for  daily backstop (emissions) data into the system 

[BETA Web Client MATS 2.0 MP Reporting Instructions June 2023.docx](https://usepa.sharepoint.com/:w:/r/sites/CAMDITLegacyReeng/_layouts/15/Doc.aspx?sourcedoc=%7B52DA5270-E093-4DCF-AC7C-124A253A721C%7D&file=BETA%20Web%20Client%20MATS%202.0%20MP%20Reporting%20Instructions%20June%202023.docx&wdLOR=c92FC035F-4657-4B3F-95FA-8BA07BAEAD9D&action=default&mobileredirect=true)

[BETA Web Client MATS 2.0 EM Reporting Instructions June ](https://usepa.sharepoint.com/:w:/r/sites/CAMDITLegacyReeng/_layouts/15/Doc.aspx?sourcedoc=%7BFE3CF8DF-B3A9-4485-83D6-993FECFD5EC1%7D&file=BETA%20Web%20Client%20MATS%202.0%20EM%20Reporting%20Instructions%20June%202023.docx&wdLOR=cE98A897F-0D9F-4BC0-88CF-CF9C649B997F&action=default&mobileredirect=true)

[BETA Web Client MATS 2.0 EM Reporting Instructions June 2023.docx](https://usepa.sharepoint.com/:w:/r/sites/CAMDITLegacyReeng/_layouts/15/Doc.aspx?sourcedoc=%7BFE3CF8DF-B3A9-4485-83D6-993FECFD5EC1%7D&file=BETA%20Web%20Client%20MATS%202.0%20EM%20Reporting%20Instructions%20June%202023.docx&wdLOR=cF2AC24DB-6DDC-4525-B8B7-19C8FDD210E5&action=default&mobileredirect=true)

### **Schema**
```
    "DailyBackstop": {
      "title": "DailyBackstop",
      "type": "object",
      "additionalProperties": true,
      "properties": {
        "unitId": {
          "type": "string",
          "minLength": 1,
          "maxLength": 6
        },        
        "date": {
          "type": "string",
          "format": "date"
        },
        "dailyNOxEmissions": {
          "type": "number",
          "multipleOf": 0.1,
          "minimum": -999999999.9,
          "maximum": 999999999.9
        },
        "dailyHeatInput": {
          "type": "number",
          "multipleOf": 0.1,
          "minimum": -999999999.9,
          "maximum": 999999999.9
        },
        "dailyAverageNOxRate": {
          "type": [
            "number",
            "null"
          ],
          "multipleOf": 0.001,
          "minimum": -9999.999,
          "maximum": 9999.999
        },
        "dailyNOxExceedence": {
          "type": "number",
          "multipleOf": 0.1,
          "minimum": -999999999.9,
          "maximum": 999999999.9
        },
        "cumulativeOSNOxExceedence": {
          "type": [
            "number",
            "null"
          ],
          "multipleOf": 0.1,
          "minimum": -999999999999.9,
          "maximum": 999999999999.9
        }
      },
      "required": [
        "unitId",
        "date",
        "dailyNOxEmissions",
        "dailyHeatImput",
        "dailyAverageNOxRate",
        "dailyNOxExceedence",
        "cumulativeOSNOxExceedence"
      ]
    },
```
### **Import JSON**
```
{
        "unitId": "1A",
        "date": "2023-07-15",
        "dailyNOxEmissions": 999999999.9,
        "dailyHeatInput": 999999999.9,
        "dailyAverageNOxRate": 9999.999,
        "dailyNOxExceedence": 999999999.9,
        "cumulativeOSNOxExceedence": 999999999999.9,
    },
```
### **Export JSON**
```
{
        "id": 1,
        "unitId": "1A",
        "monitoringLocationId": "6",
        "reportingPeriodId": 120,
        "date": "2023-07-15",
        "dailyNOxEmissions": 999999999.9,
        "dailyHeatInput": 999999999.9,
        "dailyAverageNOxRate": 9999.999,
        "dailyNOxExceedence": 999999999.9,
        "cumulativeOSNOxExceedence": 999999999999.9,
        "userId": "bvick",
        "addDate": "2023-01-20T22:40:24.000Z",
        "updateDate": "2023-01-20T22:40:24.000Z",
    },
```
### **DB Table**
```
CREATE TABLE IF NOT EXISTS camdecmps.daily_backstop(
	daily_backstop_id bigint GENERATED ALWAYS AS IDENTITY (start 1 increment 1) NOT NULL,
	unit_id numeric(38,0) NOT NULL,
	op_date date NOT NULL,
	daily_noxm numeric(10,1) NOT NULL,
	daily_hit numeric(10,1) NOT NULL,
	daily_avg_noxr numeric(7,3),
	daily_noxm_exceed numeric(10,1) NOT NULL,
	cumulative_os_noxm_exceed numeric(13,1),
	mon_loc_id character varying(45) NOT NULL,
	rpt_period_id numeric(38,0) NOT NULL,
	userid character varying(25) NOT NULL,
	add_date timestamp without time zone NOT NULL,
	update_date timestamp without time zone,
	CONSTRAINT pk_daily_backstop PRIMARY KEY (daily_backstop_id),
	CONSTRAINT fk_daily_backstop_unit FOREIGN KEY (unit_id)
		REFERENCES camd.unit (unit_id) MATCH SIMPLE,
	CONSTRAINT fk_daily_backstop_monitor_location FOREIGN KEY (mon_loc_id)
		REFERENCES camdecmps.monitor_location (mon_loc_id) MATCH SIMPLE
    ON DELETE CASCADE,
	CONSTRAINT fk_daily_backstop_reporting_period FOREIGN KEY (rpt_period_id)
		REFERENCES camdecmpsmd.reporting_period (rpt_period_id) MATCH SIMPLE
);
```

### Communications 
Also as part of this task, can you create one ticket to create the structures, and another ticket to update the beta to accept the import of these versions?  

From: Worley, Christopher 
Sent: Friday, June 23, 2023 11:10 AM
To: jasonwhitehead@cvpcorp.com; Kyle Herceg <kyleherceg@cvpcorp.com>; Whitten, Dwayne <whitten.dwayne@epa.gov>
Cc: Alex Collado <alexcollado@cvpcorp.com>; janell coleman <janellcoleman@cvpcorp.com>; Heese, Michael <Heese.Michael@epa.gov>; Ibarra, Michelle <Ibarra.Michelle@epa.gov>
Subject: FW: Updated Instructions

Jason, 

The updated instructions for MP and EM are posted and complete.  Let me know what else you need to create the updated JSON schema documentation.

Thanks.

Chris Worley
202-816-2670

From: Sarah Downes <Sarah.Downes@erg.com> 
Sent: Thursday, June 22, 2023 5:03 PM
To: Worley, Christopher <worley.christopher@epa.gov>; Frushour, Charles <Frushour.Charles@epa.gov>
Cc: Scott Aber <scott.aber@erg.com>; Ann Albrecht <Ann.Albrecht@erg.com>; Dwayne Whitten <dwayne.whitten@erg.com>; Lane Giannini <lane.giannini@erg.com>; Schreifels, Jeremy <Schreifels.Jeremy@epa.gov>
Subject: RE: Updated Instructions

Chris, 

The update for Daily Backstop has been added to the Emissions instructions on CAMD’s SharePoint.

BETA Web Client MATS 2.0 EM Reporting Instructions June 2023.docx

Sarah Downes
508.340.0198

From: Worley, Christopher <worley.christopher@epa.gov> 
Sent: Thursday, June 1, 2023 3:58 PM
To: Sarah Downes <Sarah.Downes@erg.com>; Frushour, Charles <Frushour.Charles@epa.gov>
Cc: Scott Aber <Scott.Aber@erg.com>; Ann Albrecht <Ann.Albrecht@erg.com>; Dwayne Whitten <Dwayne.Whitten@erg.com>; Lane Giannini <Lane.Giannini@erg.com>; Schreifels, Jeremy <Schreifels.Jeremy@epa.gov>
Subject: RE: Updated Instructions

CAUTION: Don't open links or attachments unless you recognize the sender and know they are safe.
Thanks Sarah,  I appreciate all the hard work you and others at ERG have done. 

From: Sarah Downes <Sarah.Downes@erg.com> 
Sent: Thursday, June 1, 2023 5:50 PM
To: Worley, Christopher <worley.christopher@epa.gov>; Frushour, Charles <Frushour.Charles@epa.gov>
Cc: Scott Aber <scott.aber@erg.com>; Ann Albrecht <Ann.Albrecht@erg.com>; Dwayne Whitten <dwayne.whitten@erg.com>; Lane Giannini <lane.giannini@erg.com>
Subject: Updated Instructions

Chris, 

The Monitoring Plan and Emissions reporting instructions have been posted to CAMD’s SharePoint:

BETA Web Client MATS 2.0 MP Reporting Instructions June 2023.docx

BETA Web Client MATS 2.0 EM Reporting Instructions June 2023.docx

Thanks,
Sarah

Sarah Downes
Eastern Research Group, Inc.
508.340.0198
sarah.downes@erg.com

**Comment by fanwari1** [comment]
The following JSON can be used for testing:
If testing from UI, use the Allen S King facility.
```
{
	"orisCode": 1915,
	"monitorPlanId": "MDC-048B9BC304774A0BADA8E13D111424FF",
	"reportingPeriodId": 117,
	"year": 2022,
	"quarter": 1,
	"submissionComment": null,
	"lastUpdated": "2022-04-12T08:34:40.000Z",
	"updatedStatusFlg": null,
	"needsEvalFlag": null,
	"chkSessionId": null,
	"submissionId": 1495782,
	"submissionAvailabilityCd": "UPDATED",
	"dailyBackstopData": [
		{
			"reportingPeriodId": 117,
			"monitoringLocationId": "988",
			"unitId": "1",
			"date": "2022-01-04",
			"dailyNOxEmissions":12,
			"dailyHeatInput":122,
			"dailyAverageNOxRate":0.123,
			"dailyNOxExceedence":111,
			"cumulativeOSNOxExceedence":12
		}
	]
}
```

**Comment by mosesdeeCVP** [comment]
![Update ECMPS to Import daily backstop [GNP] (emissions) data 5395.png](https://images.zenhubusercontent.com/5f807025564a417cbcdf6d69/023e3ae0-4ef9-4687-9f25-dec23bca8181)

**Comment by mosesdeeCVP** [comment]
Daily Backstop imports successfully
